src/building/count.c

```diff
@@ -6,6 +6,71 @@
 #include "city/health.h"
 #include "figure/figure.h"
 
+static const building_type building_set_farms[] = {
```

Comment:
I understand your intentions with this, but it may become annoying to maintain later on. Though I'm not sure what the best solution would be.

Reply:
I figured this was at least easier to maintain than the previous code where it basically was coded out as multiple function calls & if statements.
But any suggestions in what I should rather do I'm open to.

Reply:
I don't have any. This isn't perfect but it will do for now!